components/script/dom/oscillatornode.rs

```diff
@@ -113,6 +115,19 @@ impl OscillatorNodeMethods for OscillatorNode {
     fn Detune(&self) -> DomRoot<AudioParam> {
         DomRoot::from_ref(&self.detune)
     }
+
+    // https://webaudio.github.io/web-audio-api/#dom-oscillatornode-type
+    fn Type(&self) -> OscillatorType {
+        self.oscillator_type.get()
+    }
+
+    // https://webaudio.github.io/web-audio-api/#dom-oscillatornode-type
+    fn SetType(&self, oscillator: OscillatorType) {
+        self.oscillator_type.set(oscillator);
```

Comment:
According to the [spec](https://webaudio.github.io/web-audio-api/#dom-oscillatornode-type) we need to throw an `InvalidStateError` when trying to set a custom type.